### ISSUE TITLE
Should KubeSpawner override Spawner's `cmd` traitlet?

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/kubespawner#227]
## TL;DR
- The default value of c.KubeSpawner.cmd hides the default value of c.Spawner.cmd, which never is used. This is good as something else would break statements within the documentation.
- A value in c.KubeSpawner.cmd will override a value in c.Spawner.cmd
- A value in c.Spawner.cmd will be used if c.KubeSpawner.cmd is undeclared

It's confusing, but works in a reasonable way...

---

Why is both Spawner and KubeSpawner, which is inheriting from Spawner, defining a cmd traitlet in the class? I figure this means that KubeSpawner will ignore whatever is set within `c.Spawner.cmd` because it has overridden it, is this correct? I realize that I'm also assuming created KubeSpawner objects would be able to access `self.cmd` within Spawner if it does not override `cmd`.

Spawner's cmd traitlet:
https://github.com/jupyterhub/jupyterhub/blob/9ea4ca364638390ee01bb5b121c87d3025168095/jupyterhub/spawner.py#L360-L374

KubeSpawner's cmd traitlet:
https://github.com/jupyterhub/kubespawner/blob/27056a79580df957d0cc2d7b61ecfa4ce694e4fd/kubespawner/spawner.py#L185-L203

**Comment by consideRatio** [comment]
I realized how to try this out quickly and found that an unset `c.KubeSpawner.cmd` allowed `c.Spawner.cmd` to be used as fallback. But the default value for `c.Spawner.cmd` which is `['jupyterhub-singleuser']` never came into play, instead a default value of `[]` made me surprised. Where did that come from?

**Comment by consideRatio** [comment]
If
